### PR TITLE
Remove OpenSSL.

### PR DESCRIPTION
**Pull request by reitermarkus**
Closes https://github.com/rust-embedded/cross/issues/229.

### CODE REVIEW

**Comment by Emilgardis** [comment]
Shouldnt this get more attention before merging? It's a rather large change

**Comment by therealprof** [comment]
@Emilgardis I agree. This should have been RFCed so people would know about the reasoning and implications and be able to object.

**Comment by Dylan-DPC** [comment]
bors: r-

**Comment by Dylan-DPC** [comment]
There was a proposal made last year: https://github.com/rust-embedded/cross/issues/229. @Emilgardis @therealprof do you have any issues with this?

**Comment by reitermarkus** [comment]
The main points are already mentioned in the issue. For me, the most important argument is that this would get rid of a bunch of issues we wouldn't have to deal with anymore. Effectively, I'm the only active maintainer at the moment, so the fewer things to maintain the better. Also, as mentioned in the issue some crates expect OpenSSL *not* to be installed.

**Comment by Dylan-DPC** [comment]
> Effectively, I'm the only active maintainer at the momen

not true ;)

**Comment by reitermarkus** [comment]
I meant active in the sense of lines of code. Of course I appreciate you all reviewing those lines. 😉

**Comment by Emilgardis** [comment]
I have no issue with this, I think removing OpenSSL is a good idea, but as @therealprof mentioned, making it more formal is imo a good thing, this is essentially a breaking change that _could_ affect users. They should have a chance to voice their opinions. Merging after 1 hour of first approval seems wrong to me. If there's no concerns voiced after x amount of time I think it's fair to say it's safe to merge this however. :+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- Removed OpenSSL from all images.
```
Comment (therealprof) [inline CHANGELOG.md]: This is a breaking change so it should be tagged as such.

**Comment by therealprof** [comment]
As mentioned in #229 I'm in theory fine with this but I haven't had time to review the PR in detail. Also it would be great if someone could summarise the implications.

If we release this we also need to make sure to bump the minor version.

**Comment by reitermarkus** [comment]
> Also it would be great if someone could summarise the implications.

There really is only one: Users will have to install OpenSSL themselves if they need it.

File: CHANGELOG.md (hunk 1 above)
Comment (reitermarkus) [reply CHANGELOG.md]: Yeah, this will definitely be released as 0.2.0.

**Comment by reitermarkus** [comment]
Also another reason: OpenSSL 1.0.2 is EOL at the end of the year. I tried bumping it to 1.1.1 but a bunch of images are failing to build with it and I'm not going to be the one fixing them.

**Comment by therealprof** [comment]
Sorry for only noticing this now:

> I meant active in the sense of lines of code. Of course I appreciate you all reviewing those lines. 😉

I think this goes without saying but this is an official Embedded WG tools team project so this is a team effort and there's no "I" in team. Also our rules **do** apply.

**Comment by reitermarkus** [comment]
> I think this goes without saying but this is an official Embedded WG tools team project so this is a team effort and there's no "I" in team. 

You're reading too much into what I am saying. I simply stated a fact, which is that `cross` isn't a priority for anyone else on the tools team at the moment.

**Comment by therealprof** [comment]
> I simply stated a fact, which is that cross isn't a priority for anyone else on the tools team at the moment.

This may or may not be true but is completely irrelevant. I just wanted to point out that the team is the owner and maintainer of this crate and we do have rules and a code of conduct to adhere to.

**Comment by reitermarkus** [comment]
> completely irrelevant

This is completely relevant. We can't keep OpenSSL support if there isn't anyone to actively maintain it.

**Comment by therealprof** [comment]
@reitermarkus 

I'm not objecting to removing OpenSSL just trying to keep focus on the real discussion so an informed decision can be made by the **team**.

I am objecting to some of the comments which have been made by pointing out that this is a an official project by the Embedded WG and not a one/two man show.

**Comment by therealprof** [comment]
@adamgreig @ryankurte @Disasm @burrbull 

Your opinion and approval is requested here.

**Comment by Disasm** [comment]
I don't understand how cross works, so I can't estimate implications of this PR. As digging into details requires too much effort and time from my side (which I do not have atm), I would prefer to abstain from voting.

**Comment by jamesmunns** [comment]
The discussion here seems to be going a little beyond just OpenSSL, but I wanted to add a couple points.

First: The `cross` project was severely neglected before you joined the tools team @reitermarkus. @Dylan-DPC and myself were contributors on the `cross` project before it was taken over by the Tools team, but I admit I haven't had the bandwidth to help out for a while. I am extremely happy to see Cross being maintained right now, and I want to thank you and Dylan for picking it up over the past weeks.

Second, as this project is owned by the Tools teams, as @therealprof mentions, our policies apply here for the maintenance. The two relevant policies I can see are:

* [The Reviewer Handbook](https://github.com/rust-embedded/wg/blob/master/ops/review.md)
* [The Publishing Workflow](https://github.com/rust-embedded/wg/blob/master/ops/publish.md)

In order to meet the policies above, I would suggest that we:

1. Either add @Dylan-DPC to the Tools team (through the /wg process), or get an explicit :+1: from the whole Tools team to have him maintain only the `cross` crate. 
2. @therealprof has already called in the rest of the tools team, so that fits the needs of review by the tools for this breaking change

Additionally if possible, I think we could make this breaking change lower friction if we could document how to correctly add OpenSSL after this PR has been merged.

**Comment by adamgreig** [comment]
I've been staying out of this thread so far since I'm not really involved in cross. Like James I'm pleased to see it getting so much attention! As far as I can tell this is a fairly normal breaking change, the sort of thing that wouldn't typically require an RFC but just a regular approval process. I don't see any issue removing OpenSSL, for all the good reasons already mentioned in the thread. I'm happy to approve the PR.

In terms of team policy I appreciate we should follow our procedures though I don't think any egregious violation has been committed here. I'm very happy for @Dylan-DPC to join the tools team if they want, or to just continue maintaining cross with approval from the tools team. One or the other should happen to keep us in line with our policies.

File: CHANGELOG.md (hunk 1 above)
Comment (therealprof) [reply CHANGELOG.md]: By tagging I meant it should be mentioned in the ChangeLog. But that can be done before the release, too.

**Comment by therealprof** [comment]
... and we have a majority of votes. Thanks for weighing in everybody.

**Comment by Dylan-DPC** [comment]
@therealprof thanks for the help. I understand your point here. However, given that the tools team is busy on other projects and life and the staleness that cross has had for the most of this year, it is difficult to wait for the entire team to express their concerns on PRs.

**Comment by therealprof** [comment]
@Dylan-DPC We're working on building up the team such as bringing in @reitermarkus. In the last few months we've welcomed a few additional members and we're happy to welcome more people interested in helping out.

But it is what it is, this project has been brought into the WG and maintained under our umbrella so everything happening on this project has to happen in compliance with the WG rules, no matter how 

> difficult to wait for the entire team to express their concerns on PRs

it is. Quite a few people are now relying on `cross` and also we as WG are committed to delivering quality. This cannot be achieved by skipping a proper review process or the consultation with the rest of the team for breaking changes; however cumbersome or daunting it may be.

**Pull request merged**